packages/polling-controller/src/PollingController.ts

```diff
@@ -1,150 +1,144 @@
-import { BaseControllerV2 } from '@metamask/base-controller';
-import type {
-  RestrictedControllerMessenger,
-  StateMetadata,
-} from '@metamask/base-controller';
+import { BaseController, BaseControllerV2 } from '@metamask/base-controller';
 import type { NetworkClientId } from '@metamask/network-controller';
-import type { Json } from '@metamask/utils';
 import { v4 as random } from 'uuid';
 
-export type PollingCompleteType<N extends string> = {
-  type: `${N}:pollingComplete`;
-  payload: [string];
-};
+// eslint-disable-next-line @typescript-eslint/ban-types
+type Constructor = new (...args: any[]) => {};
```

Comment:
Nice to have a comment here to explain why we're using `any`.

Also, `object` is the same as `{}` except it isn't banned by our ESLint rules
```suggestion
// Mixin classes require a constructor with an `...any[]` parameter
// See TS2545
type Constructor = new (...args: any[]) => object;
```